public/app/features/dashboard/components/ShareModal/SharePublicDashboard/SharePublicDashboard.tsx

```diff
@@ -88,21 +104,21 @@ export const SharePublicDashboard = (props: Props) => {
   }, [props.dashboard.events, forceUpdate]);
 
   useEffect(() => {
-    if (publicDashboardPersisted(publicDashboard)) {
-      setAcknowledgements({
-        public: true,
-        datasources: true,
-        usage: true,
-      });
-      setAnnotationsEnabled(!!publicDashboard?.annotationsEnabled);
-    }
-
-    setEnabledSwitch((prevState) => ({ ...prevState, isEnabled: !!publicDashboard?.isEnabled }));
-  }, [publicDashboard]);
+    const isPublicDashboardPersisted = publicDashboardPersisted(publicDashboard);
+    reset({
+      publicAcknowledgment: isPublicDashboardPersisted,
+      dataSourcesAcknowledgment: isPublicDashboardPersisted,
+      usageAcknowledgment: isPublicDashboardPersisted,
+      isAnnotationsEnabled: publicDashboard?.annotationsEnabled,
+      enabledSwitch: publicDashboard?.isEnabled,
+    });
+  }, [publicDashboard, reset]);
 
   const isLoading = isGetLoading || isSaveLoading || isUpdateLoading;
   const hasWritePermissions = contextSrv.hasAccess(AccessControlAction.DashboardsPublicWrite, isOrgAdmin());
-  const acknowledged = acknowledgements.public && acknowledgements.datasources && acknowledgements.usage;
+  const acknowledged =
+    watch('publicAcknowledgment') && watch('dataSourcesAcknowledgment') && watch('usageAcknowledgment');
```

Comment:
`watch()` gets the current form value?

Reply:
correct! anytime the value changes, I can get that value with that method